File: my-folder/3371-harshad-number/solution.py
class Solution:
    def sumOfTheDigitsOfHarshadNumber(self, x: int) -> int:
        digits = [char for char in str(x)]
        s = 0
        for char in digits:
            s+=int(char)

        if x%s==0:
            return s
        else:
            return -1

File: my-folder/3371-harshad-number/solution.cpp
class Solution {
public:
    int sumOfTheDigitsOfHarshadNumber(int x) {
        std::vector<int> digits;
        int temp = x;
        while (temp)
        {
            int digit = temp%10;
            digits.push_back(digit);
            temp/=10;
        }
        int s = std::accumulate(digits.begin(), digits.end(), 0);
        return (x % s == 0) ? s : -1;
    }
};
